compute/cloud-client/src/main/java/compute/snapshotschedule/EditSnapshotSchedule.java

```diff
@@ -0,0 +1,119 @@
+
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Operation.Status;
+import com.google.cloud.compute.v1.PatchResourcePolicyRequest;
+import com.google.cloud.compute.v1.ResourcePoliciesClient;
+import com.google.cloud.compute.v1.ResourcePolicy;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicy;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicyRetentionPolicy;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicySchedule;
+import com.google.cloud.compute.v1.ResourcePolicySnapshotSchedulePolicySnapshotProperties;
+import com.google.cloud.compute.v1.ResourcePolicyWeeklyCycle;
+import com.google.cloud.compute.v1.ResourcePolicyWeeklyCycleDayOfWeek;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class EditSnapshotSchedule {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+    // Name of the region where your snapshot schedule is located.
+    String region = "us-central1";
+    // Name of the snapshot schedule you want to update.
+    String snapshotScheduleName = "YOUR_SCHEDULE_NAME";
+
+    editSnapshotSchedule(projectId, region, snapshotScheduleName);
+  }
+
+  // Edits a snapshot schedule.
+  public static Status editSnapshotSchedule(
+          String projectId, String region, String snapshotScheduleName)
+          throws IOException, InterruptedException, ExecutionException, TimeoutException {
+
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (ResourcePoliciesClient resourcePoliciesClient = ResourcePoliciesClient.create()) {
+      Map<String, String> snapshotLabels = new HashMap<>();
+      snapshotLabels.put("key", "value");
+
+      ResourcePolicySnapshotSchedulePolicySnapshotProperties.Builder snapshotProperties =
+              ResourcePolicySnapshotSchedulePolicySnapshotProperties.newBuilder();
+      snapshotProperties.putAllLabels(snapshotLabels);
+
+      ResourcePolicyWeeklyCycleDayOfWeek dayOfWeek = ResourcePolicyWeeklyCycleDayOfWeek.newBuilder()
+              .setDay("Tuesday")
+              .setStartTime("09:00")
+              .build();
+      ResourcePolicyWeeklyCycle weeklySchedule = ResourcePolicyWeeklyCycle.newBuilder()
+              .addDayOfWeeks(dayOfWeek)
+              .build();
+
+      ResourcePolicySnapshotSchedulePolicySchedule.Builder scheduler =
+              ResourcePolicySnapshotSchedulePolicySchedule.newBuilder();
+      scheduler.clearDailySchedule().clearHourlySchedule();
```

Comment:
This can be removed as the scheduler is constructed from a new builder.

Reply:
Thank you! Fixed